components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMServiceTest.java

```diff
@@ -213,16 +265,136 @@ public void getApplicationDTOTest() throws Exception {
+        try {
+            dcrmService.getApplication(dummyConsumerKey);
+        } catch (DCRMException ex) {
+            assertEquals(ex.getMessage(), String.format(DCRMConstants.ErrorMessages.
+                    TENANT_DOMAIN_MISMATCH.getMessage(), dummyConsumerKey));
+            return;
+        }
+        fail("Expected exception IdentityException not thrown by getApplication method");
+
+    }
+
+    @Test
+    public void validateRequestTenantDomainTestWitIdentityOAuth2Exception()
+            throws IdentityOAuth2Exception, InvalidOAuthClientException {
+
+        when(OAuth2Util.getTenantDomainOfOauthApp(dummyConsumerKey)).thenThrow(new IdentityOAuth2Exception(""));
+        try {
+            dcrmService.getApplication(dummyConsumerKey);
+        } catch (DCRMException ex) {
+            assertEquals(ex.getMessage(), String.format(DCRMConstants.ErrorMessages.
+                    FAILED_TO_VALIDATE_TENANT_DOMAIN.getMessage(), dummyConsumerKey));
+            return;
+        }
+        fail("Expected exception IdentityException not thrown by getApplication method");
+
+    }
+
+    @Test
+    public void getApplicationByNameTest() throws Exception {
+
+        startTenantFlow();
+        OAuthConsumerAppDTO oAuthConsumerApp = new OAuthConsumerAppDTO();
+        oAuthConsumerApp.setApplicationName(dummyClientName);
+        String dummyConsumerSecret = "dummyConsumerSecret";
+        oAuthConsumerApp.setOauthConsumerSecret(dummyConsumerSecret);
+        oAuthConsumerApp.setOauthConsumerKey(dummyConsumerKey);
+        String dummyCallbackUrl = "dummyCallbackUrl";
+        oAuthConsumerApp.setCallbackUrl(dummyCallbackUrl);
+        oAuthConsumerApp.setUsername(dummyUserName.concat("@").concat(dummyTenantDomain));
+
+        when(mockApplicationManagementService.getServiceProvider(anyString(), anyString()))
+                .thenReturn(new ServiceProvider());
+        when(mockOAuthAdminService
+                .getOAuthApplicationDataByAppName(dummyClientName)).thenReturn(oAuthConsumerApp);
+        Whitebox.setInternalState(dcrmService, "oAuthAdminService", mockOAuthAdminService);
+        Application application = dcrmService.getApplicationByName(dummyClientName);
+
+        assertEquals(application.getClientId(), dummyConsumerKey);
+        assertEquals(application.getClientName(), dummyClientName);
+        assertEquals(application.getClientSecret(), dummyConsumerSecret);
+        assertEquals(application.getRedirectUris().get(0), dummyCallbackUrl);
+    }
+
+    @Test
+    public void getApplicationEmptyClientNameTest() throws DCRMException {
+
+        try {
+            dcrmService.getApplicationByName("");
+        } catch (IdentityException ex) {
+            assertEquals(ex.getErrorCode(), DCRMConstants.ErrorMessages.BAD_REQUEST_INSUFFICIENT_DATA.toString());
+            return;
+        }
+        fail("Expected exception IdentityException not thrown by getApplication method");
```

Comment:
```suggestion
        fail("Expected IdentityException was not thrown by getApplication method");
```
Correct in other places as well